src/kbmod/search/KBMOSearch.cpp

```diff
@@ -340,40 +340,47 @@ std::vector<RawImage> KBMOSearch::coaddedScienceStampsGPU(std::vector<trajectory
     perImageData img_data;
     img_data.numImages = num_images;
     img_data.imageTimes = stack.getTimesDataRef();
-    if (params.useCorr) img_data.baryCorrs = &baryCorrs[0];
 
     // Allocate space for the results.
     const int num_trajectories = t_array.size();
-    const int stamp_width = 2 * radius + 1;
+    const int stamp_width = 2 * params.radius + 1;
     const int stamp_ppi = stamp_width * stamp_width;
     std::vector<float> stamp_data(stamp_ppi * num_trajectories);
 
     // Do the co-adds.
-    deviceGetCoadds(stack, img_data, radius, compute_mean, num_trajectories, t_array.data(),
-                    use_index_vect, stamp_data.data());
+    deviceGetCoadds(stack, img_data, num_trajectories, t_array.data(), params,
+                    use_index_vect, stamp_data.data()); 
 
     // Copy the stamps into RawImages
     std::vector<RawImage> results(num_trajectories);
     std::vector<float> current_pixels(stamp_ppi, 0.0);
+    std::vector<float> empty_pixels(1, NO_DATA);
     for (int t = 0; t < num_trajectories; ++t) {
+        bool all_no_data = true;
         int offset = t * stamp_ppi;
         for (unsigned p = 0; p < stamp_ppi; ++p) {
             current_pixels[p] = stamp_data[offset + p];
+            all_no_data = all_no_data && (stamp_data[offset + p] == NO_DATA);
+        }
+
+        if (all_no_data && params.do_filtering) {
+            results[t] = RawImage(1, 1, empty_pixels);
+        } else {
+            results[t] = RawImage(stamp_width, stamp_width, current_pixels);
         }
-        results[t] = RawImage(stamp_width, stamp_width, current_pixels);
     }
     return results;
 }
 
 std::vector<RawImage> KBMOSearch::coaddedScienceStampsGPU(std::vector<trajectory>& t_array,
-                                                          int radius, bool compute_mean) {
+                                                          stampParameters& params) {
```

Comment:
If params is read only then consider making it a const &

Reply:
Done